pkg/api/login.go

```diff
@@ -78,12 +78,13 @@ func tryOAuthAutoLogin(c *m.ReqContext) bool {
 	return false
 }
 
-func (hs *HTTPServer) LoginAPIPing(c *m.ReqContext) Response {
-	if c.IsSignedIn || c.IsAnonymous {
-		return JSON(200, "Logged in")
+func (hs *HTTPServer) LoginAPIPing(c *m.ReqContext) {
+	if c.IsSignedIn || (c.AllowAnonymous && c.IsAnonymous) {
```

Comment:
can `c.IsAnonymous` be `true` unless `c.AllowAnonymous` is configured?

Reply:
Any reason why this sas changed from returning Response to doing c.JsonApiError?

Reply:
Because of "Fixes issue with /api/login/ping returning 200 OK even though a 401 unauthorized was intended.". What's the problem with using JsonApiError?

Reply:
because it's the old way, for methods that have not been converted to return Response. 

> Because of "Fixes issue with /api/login/ping returning 200 OK even though a 401 unauthorized was intended.". 

So there is a bug in Error(401, "Unauthorized", nil), that sounds like a bug that should be fixed, that is used everywhere else

Reply:
Yeah that's a good point. Didn't think about that. Just looked at git blame and used last commit before we changed it to Response. Will investigate why it doesn't work

Reply:
Yeah that's a good point. Didn't think about that. Just looked at git blame and used last commit before we changed it to Response. Will investigate why it doesn't work

Reply:
Yeah that's a good point. Didn't think about that. Just looked at git blame and used last commit before we changed it to Response. Will investigate why it doesn't work